### DISCUSSION TITLE
How to get the flame height in FDS

### DISCUSSION THREAD

**Discussion by hunzideriji** [firemodels/fds#13155] [Q&A] [answered]
Hello, I am simulating the combustion of propane gas through a burner, how can I obtain the flame height of propane combustion?

**Comment by drjfloyd** [comment]
Have you tried searching the User's Guide?

**Reply by hunzideriji** [reply]
Dear Dr Floyd,
Thanks a lot for your help.
But when I built a simple fire source model and used the method in the manual to solve the flame height, the result was zero before 1.5s, and then it started to show after 1.5s. What is the reason? Below is my code and my result file.
[新建文本文档 (2).txt](https://github.com/user-attachments/files/16193171/2.txt)
[1737095837_devc.csv](https://github.com/user-attachments/files/16193175/1737095837_devc.csv)

**Reply by drjfloyd** [reply, marked as answer]
See 22.2.5 Linear Array of Point Devices

**Reply by drjfloyd** [reply]
A single “line” file can hold more than a single line of data. By default, the coordinate columns are
labeled using the ID of the DEVC appended with either -x, -y, or -z. To change these labels, use X_ID,
Y_ID, and/or Z_ID. To suppress the coordinate columns altogether, add HIDE_COORDINATES=T to the
DEVC line. This is convenient if you have multiple arrays of data that use the same coordinates.

**Reply by hunzideriji** [reply]
Thank you for your help

**Reply by drjfloyd** [reply]
The User's Guide is searchable.  That is the first instance of Z_ID if you search from the start of the document.

**Reply by obscureed** [reply]
It does seem that the L_F flame-height result only becomes available halfway through the simulation. This is strange. @drjfloyd, any thoughts?

Here's an example with T_END=4 (but I also tried T_END=8, and the flame height is non-zero only from t=4.0 onwards).
[github13155.fds.txt](https://github.com/user-attachments/files/16194247/github13155.fds.txt)

The attached version has some small improvements over the original, I think:
• The sample points have z-coordinates corresponding to cell centres (as the User Guide suggests) -- or at least, I think it does. There are no opportunities for the coordinates to be displayed in this case.
• The propane injection corresponds exactly to cell faces.
• The unnecessary ignitor has been removed.
• The unnecessary stuff below z=0 has been removed.

**Reply by drjfloyd** [reply]
@obscureed  

Line devices were impleneted for the primary use case of getting time-averaged, steady values:

22.2.5 Linear Array of Point Devices

In a file called CHID_line.csv,...time-averaged over the interval between STATISTICS_START and STATISTICS_END. The default value
of the latter is T_END and the former is half the total simulation time.

**Comment by Sadeghi6872** [comment]
Dear Dr Floyd, 
Thanks a lot for your help. 
Best regards,
Mohamad
